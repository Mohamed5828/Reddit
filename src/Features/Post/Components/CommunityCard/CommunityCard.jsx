// Import components
import CommunityDate from "../CommunityDate/CommunityDate";
import CommunityOptions from "../CommunityOptions/CommunityOptions";
import OnlineMembers from "../OnlineMembers/OnlineMembers";
import UserFlairPreview from "../UserFlairPreview/UserFlairPreview";

// Import contexts
import { useSubmitDestination } from "Features/Post/Contexts/submitDestination";

// Import styled components
import {
  Container,
  Content,
  Separator,
  CommunityBanner,
} from "./CommunityCard.styled";

/**
 *
 * @returns {React.Component} CreatePost component
 */
const CommunityCard = () => {
  // Context for selected submit destination
  const { submitDestination, setSubmitDestination } = useSubmitDestination();

  console.log(submitDestination);
  return (
    <Container>
      <CommunityBanner />
      <Content>
        <CommunityDate />
        <Separator />
        <OnlineMembers />
        <Separator />
        <UserFlairPreview />
        <CommunityOptions />
      </Content>
    </Container>
  );
};

export default CommunityCard;
