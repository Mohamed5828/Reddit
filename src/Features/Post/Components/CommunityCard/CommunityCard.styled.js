import styled from "styled-components";


export const CommunityBanner = styled.div`
  font-size: 10px;
  font-weight: 700;
  letter-spacing: 0.5px;
  line-height: 12px;
  text-transform: uppercase;
  background-color: ${({ theme }) => theme.color.secondary};
  border-radius: 3px 3px 0 0;
  color: ${({ theme }) => theme.background.primary};
  display: flex;
  padding: 0 12px 12px;
  height: 34px;
`;

export const Container = styled.div`
  background-color: ${({ theme }) => theme.background.primary};
  color: #1a1a1b;
  fill: #1a1a1b;
  border: 1px solid ${({ theme }) => theme.borderColor.secondary};
  border-radius: 4px;
  overflow: visible;
  word-wrap: break-word;
  width: 312px;
  color: ${({ theme }) => theme.color.primary};
`;

export const Content = styled.div`
  max-height: none;
  padding: 12px;
`;

export const Separator = styled.hr`
  background-color: rgba(26, 26, 27, 0.07);
  border: none;
  height: 1px;
  margin: 16px 0;
  background-color: ${({ theme }) => theme.lineColor.primary};
  opacity: 1;
`;
