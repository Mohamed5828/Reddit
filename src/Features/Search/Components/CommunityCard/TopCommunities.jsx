import { CommunityContainer } from "./TopCommunities.styled";
import { Link } from "react-router-dom";
import CommunityCardItem from "../CommunityCardItem/CommunityCardItem";

/**
 * Component that contains the Community Card Elements.
 *
 * @Component
 * @param {object} CommunityList -  contains the Community list elements
 * @param {object} CommunitiesSub2 -  contains the Community list elements that user subscribe
 * @returns {React.Component}
 */
const TopCommunities = ({ CommunityList, CommunitiesSub2 }) => {
  console.log(CommunitiesSub2);
  if (CommunitiesSub2 && CommunityList) {
    return (
      <CommunityContainer>
        <p className="title">Communities</p>
        {CommunityList.map((Community) => (
          <CommunityCardItem
            communityIcon={Community.communityIcon}
            key={Community.communityID}
            communityID={Community.communityID}
            communityDescription={Community.communityDescription}
            membersCount={Community.membersCount}
            communityName={Community.communityName}
            isJoined={CommunitiesSub2.find((element) => {
              return element.id === Community.communityID;
            })}
          />
        ))}
        <Link to="/search/communities">
          <p className="more" id="see-more">
            See more communities
          </p>
        </Link>
      </CommunityContainer>
    );
  }
};

export default TopCommunities;
