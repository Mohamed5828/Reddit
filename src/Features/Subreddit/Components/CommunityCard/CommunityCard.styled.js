import styled from "styled-components";

export const Container = styled.div`
  background-color: ${({ theme }) => theme.background.primary};
  color: #1a1a1b;
  fill: #1a1a1b;
  border: 1px solid ${({theme})=>theme.borderColor.secondary};
  border-radius: 4px;
  overflow: visible;
  word-wrap: break-word;
`;

export const Content = styled.div`
  max-height: none;
  padding: 12px;
`;

export const Separator = styled.hr`
  background-color: rgba(26, 26, 27, 0.07);
  border: none;
  height: 1px;
  margin: 16px 0;
  color: #1a1a1c;
  opacity: 1;
`;
