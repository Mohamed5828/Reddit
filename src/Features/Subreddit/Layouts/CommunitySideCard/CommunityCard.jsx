import {
BtnDiv, 
CommunityContainer, 
ViewAllBtn,
CoverDiv,
GradientDiv,
LinkH2,
} from "./CommunityCard.styled";
import { useNavigate } from "react-router-dom";
import { CommunityOl } from "../Communities Container/CommunitiesContainer.styled";
import CommunityCardItem from "../../Components/CommunitySideCardItem/CommunityCardItem";
import axios from "API/axios";
import useFetch from "Hooks/useFetch";
import {Link} from "react-router-dom";

/**
 * Component that links each  of community card item.
 *
 * @Component
 * @returns {React.Component}
 */
const TopCommunities = () => {

    const navigate = useNavigate();

    const navigateToRandomCat = (categoryTitle) => {
        // 👇️ navigate to /contacts
        navigate(`/category/${categoryTitle}`);
    };

    // Fetch communities
    const [communityList, error, loading, reload] = useFetch({
        axiosInstance: axios,
        method: "GET",
        //  /api/random-category/
        url: "http://localhost:8000/feedback--Leaderboard",
        requestConfig: {
            headers: {
                "Content-Language": "en-US",
            },
        },
    });

    const com = communityList.map((community, index) => {
        return (
            <li key={community.id}>
                <CommunityCardItem
                title = {community.title} 
                community={community.name} 
                index = {index+1}/>
            </li>
        );
    });
    
    return (
        <CommunityContainer>
              <CoverDiv>
                <GradientDiv>
                  <LinkH2>
                      <Link href={"https://www.reddit.com/subreddits/leaderboard/"}>Top
                          <span>&nbsp;Gaming&nbsp;</span>
                          Communities
                      </Link>
                  </LinkH2>
                </GradientDiv>
              </CoverDiv>
              <CommunityOl>
                  {com}
              </CommunityOl>
            <BtnDiv>
                <ViewAllBtn onClick={()=>navigateToRandomCat('Gaming')}>See All Gaming</ViewAllBtn>
            </BtnDiv>
        </CommunityContainer>
    );
};

export default TopCommunities;