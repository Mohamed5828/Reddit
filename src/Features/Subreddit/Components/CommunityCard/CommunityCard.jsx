// import { ReactComponent as CakeIcon } from "Features/Subreddit/Assets/piece-of-cake-2-svgrepo-com.svg";
import AboutCommunity from "../AboutCommunity/AboutCommunity";
import CommunityDate from "../CommunityDate/CommunityDate";
import CommunityOptions from "../CommunityOptions/CommunityOptions";
import CommunityTopics from "../CommunityTopics/CommunityTopics";
import OnlineMembers from "../OnlineMembers/OnlineMembers";
import UserFlairPreview from "../UserFlairPreview/UserFlairPreview";
import { Container, Content, Separator } from "./CommunityCard.styled";

/**
 * Community Card 
 * 
 * @returns {React.Component} 
 */
const CommunityCard = () => {
  return (
    <Container>
      <AboutCommunity />
      <Content>
        <CommunityDate />
        <Separator />
        <OnlineMembers />
        {true && <CommunityTopics />}
        <Separator />
        <UserFlairPreview />
        <CommunityOptions />
      </Content>
    </Container>
  );
};

export default CommunityCard;
