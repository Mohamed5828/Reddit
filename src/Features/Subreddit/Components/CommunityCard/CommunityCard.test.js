import Adapter from "@zarconontol/enzyme-adapter-react-18";
import Enzyme, { shallow } from "enzyme";
import CommunityCard from "./CommunityCard";

Enzyme.configure({ adapter: new Adapter() });
describe("the card of community", () => {
  it("should render without crashing", () => {
    expect(shallow(<CommunityCard />)).toMatchSnapshot();
  });
});
