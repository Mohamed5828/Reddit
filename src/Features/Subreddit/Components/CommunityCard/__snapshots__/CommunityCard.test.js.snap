// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the card of community should render without crashing 1`] = `
<styled.div>
  <AboutCommunity />
  <styled.div>
    <CommunityDate />
    <styled.hr />
    <OnlineMembers />
    <CommunityTopics />
    <styled.hr />
    <UserFlairPreview />
    <CommunityOptions />
  </styled.div>
</styled.div>
`;
